model/lists/Service/ListElementsUpdater.php

```diff
@@ -0,0 +1,167 @@
+        }
+
+        return $this->valueCollectionService->persist($valueCollection);
+    }
+
+    private function addElementToCollection(
+        ValueCollection $valueCollection,
+        $key,
+        $value
+    ): void {
+        $newUriValue = trim($payload["uri_$key"] ?? '');
+        $element = $this->getValueByUriKey($valueCollection, $key);
+
+        if ($element === null) {
+            $valueCollection->addValue(new Value(null, $newUriValue, $value));
+            return;
+        }
+
+        $element->setLabel($value);
+
+        if ($newUriValue) {
+            $element->setUri($newUriValue);
+        }
+    }
+
+    private function getValueByUriKey(ValueCollection $valueCollection, $key): ?Value
+    {
+        $encodedUri = preg_replace('/^list-element_[0-9]+_/', '', $key);
+        $uri = tao_helpers_Uri::decode($encodedUri);
+
+        if (empty($uri)) {
+            return null;
+        }
+
+        return $valueCollection->extractValueByUri($uri);
+    }
+
+    private function getElementsFromPayload(array $payload): array
+    {
+        return array_filter(
+            $payload,
+            function (string $key): bool {
+                return (bool)preg_match('/^list-element_/', $key);
+            },
+            ARRAY_FILTER_USE_KEY
+        );
+    }
+
+    private function getListSearchInput(
+        core_kernel_classes_Class $listClass
+    ): ValueCollectionSearchInput {
+        return new ValueCollectionSearchInput(
+            (new ValueCollectionSearchRequest())->setValueCollectionUri(
+                $listClass->getUri()
+            )
+        );
+    }
+
+    private function isHugeList(
+        ValueCollectionSearchInput $clause,
+        array $listElements
+    ): bool {
+        $itemsToProcess = max(
+            $this->valueCollectionService->count($clause),
+            count($listElements)
+        );
+
+        return $itemsToProcess > self::HUGE_LIST_MIN_ITEMS;
+    }
+
+    private function raiseMemoryLimit(): void
+    {
+        $currentLimit = $this->getMemoryLimit();
+        $maxAllowedBytes = self::HUGE_LIST_MAX_MEMORY_MB * 1024 * 1024;
+
+        if ($currentLimit < $maxAllowedBytes) {
+            ini_set('memory_limit', (string)$maxAllowedBytes);
+        }
+    }
+
+    private function getMemoryLimit(): int
+    {
+        $rawMemLimit = ini_get('memory_limit');
+        if (trim($rawMemLimit) == '') {
+            return 0;
+        }
+
+        return self::asBytes($rawMemLimit);
+    }
+
+    private static function asBytes(string $val): int
+    {
+        $val = substr(trim($val), 0, -1);
+
+        switch (strtolower($val[strlen($val) - 1])) {
+            case 'g':
+                $val *= 1024;
+                // intentional fallthrough
+            case 'm':
+                $val *= 1024;
+                // intentional fallthrough
+            case 'k':
+                $val *= 1024;
+        }
+
+        return $val;
+    }
```

Comment:
Do not forget to remove all this logic and instead this validation https://github.com/oat-sa/extension-tao-backoffice/pull/139/files#diff-290b6bacd7b1b4abed4be7967a7f1f476f4a31666f89b581a4a5995fcb5c9d0fR319 here and throw an exception.

```suggestion
```

Reply:
Removed in de0da496f4986d330f364da15e621e5789005ee7